packages/backend-plugin-api/src/wiring/createBackendFeatureLoader.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ServiceRef } from '../services';
+import { BackendFeature } from '../types';
+import { describeParentCallSite } from './describeParentCallSite';
+import { InternalBackendFeatureLoader } from './types';
+
+/**
+ * @public
+ * Options for creating a new backend feature loader.
+ */
+export interface CreateBackendFeatureLoaderOptions<
+  TDeps extends { [name in string]: unknown },
+> {
+  deps?: {
+    [name in keyof TDeps]: ServiceRef<TDeps[name], 'root'>;
+  };
+  loader(
+    deps: TDeps,
```

Comment:
This is a bit interesting to me from an inference standpoint, like what happens if you do specify a deps but ignore it in the loader (zero arguments) etc. But it's probably fine!

Reply:
Yep omitting the param is fine too. If you don't define `deps` then you'll end up with `Record<string, unknown>` as the first param, which I think is fine too.